### ISSUE TITLE
n2 diagram breaks when class names contain 'nan' and other letters before and after

### ISSUE THREAD

**Issue by cfe316** [OpenMDAO/OpenMDAO#1997]
### Summary of Issue

N2 diagram fails for class names containing nan or Nan

### Issue Type

- [x] Bug
- [ ] Enhancement  
      POEM #___ <N/A or POEM number at https://github.com/OpenMDAO/POEMs>

### Description

The n2 diagram does not load when I have a class name that contains either `*nan` or `nan*` or `*nan*`, where * is one or more letters. The bug does not appear to be sensitive to capitalization (I only tried changing the first letter though). I originally found this when I wanted to name a class `OperationsAndMaintenanceCost`, which I can rename to `OMCosts` easily enough, but there is serious tokamak physics that has the word `Banana` in it which I might like to describe someday.

When this bug occurs, the whole n2 screen comes up error-red and the diagram does not load. Of the normal diagram one 'pull-outer' arrow appears but that's all. There's no friendly n2 error message either.

This happens in both Firefox and in Chromium.

### Example
```
import openmdao.api as om

# breaks
class Banana(om.ExplicitComponent):
    pass

# breaks
class IpodNano(om.ExplicitComponent):
    pass

# breaks
class Nanana(om.ExplicitComponent):
    pass

# breaks
class Bananas(om.ExplicitComponent):
    pass

# breaks
class Ananas(om.ExplicitComponent):
    pass

# breaks
class Dynanite(om.ExplicitComponent):
    pass

# okay
class Montana(om.ExplicitComponent):
    pass

# okay
class Banunu(om.ExplicitComponent):
    pass

# okay
class Banan(om.ExplicitComponent):
    pass

# okay
class Nano(om.ExplicitComponent):
    pass

# okay
class Nan(om.ExplicitComponent):
    pass

# okay
class Nanite(om.ExplicitComponent):
    pass

# okay
class Dynamite(om.ExplicitComponent):
    pass

# okay
class Mamama(om.ExplicitComponent):
    pass

if __name__ == '__main__':
    prob = om.Problem()

    class test(om.Group):
        def setup(self):
            self.add_subsystem("test", Banana(),
                    promotes=["*"])

    prob.model = test()
    prob.setup()
    prob.run_driver()
```

```
openmdao n2 test.py
```

### Environment

Operating System: Ubuntu 20.04

Python environment:
Python 3.8.5 (default, Aug  3 2020, 14:44:06)
[GCC 9.3.0] on linux

Packages:

`OpenMDAO 3.8.0`

**Comment by swryan** [comment]
looks like this is addressed in an imminent PR (#1989)... should get this fix in shortly

**Comment by robfalck** [comment]
I've added this issue to the resolves list of #1989

**Comment by cfe316** [comment]
F'nan'tastic. Looking forward to the update.
